packages/nuxt/src/app/entry.ts

```diff
@@ -30,11 +31,17 @@ if (process.server) {
 
     const nuxt = createNuxtApp({ vueApp, ssrContext })
 
+    async function contextCaller (hooks: HookCallback[], args: any[]) {
+      for (const hook of hooks) {
+        await nuxtAppCtx.call(nuxt, () => hook(...args))
+      }
+    }
+
     try {
       await applyPlugins(nuxt, plugins)
-      await nuxt.hooks.callHook('app:created', vueApp)
+      await nuxt.hooks.callHookWith(contextCaller, 'app:created', vueApp)
```

Comment:
Something else i'm wondering is that we probably only need this for server-side only. It can be a breaking change for client-side as there, we have context always available assuming there is one nuxt instance. 

Also we could directly patch `nuxt.callHook` with contextCaller wdyt?

Reply:
Yes, this change only affects the server side.

As for patching, that's a nice idea. Will update.